_assets/ci/Jenkinsfile.tests-rpc

```diff
@@ -71,6 +71,9 @@ pipeline {
     }
     success { script { github.notifyPR(true) } }
     failure { script { github.notifyPR(false) } }
-    cleanup { sh 'make git-clean' }
+    cleanup {
+      sh 'make git-clean'
```

Comment:
What's the point of the `make git-clean` if you're afterwards using `cleanWs()`?

Also, isn't the issue here that files created by the tests are owned not by `jenkins` user due to running in a container?

Reply:
Okay I can remove `make git-clean`

The error message was : 

```
[2024-10-21T13:35:07.111Z] ln: failed to create symbolic link 
'/home/jenkins/workspace/go_prs_linux_x86_64_main_PR-5941@tmp/go/src/github.com/status-im/status-go': 
File exists
```

and it happened because of 
https://github.com/status-im/status-go/blob/3179532b645549c103266e007694d2c81a7091b4/_assets/ci/Jenkinsfile.linux#L56

So I assumed that this issue happens because of improper workspace cleanup during multiple runs of the same PR in Jenkins

Reply:
Look at `git blame`, we've had that symlinking since 2022:

![image](https://github.com/user-attachments/assets/36755d3c-92f1-4aee-99a7-b26ff649fa14)

Why would that specifically start causing issues now?

Reply:
Adding `cleanWs()` isn't actually solving the actual issue. Look at the path:
```
/home/jenkins/workspace/go_prs_linux_x86_64_main_PR-5941@tmp/go/src/github.com/status-im/status-go
```
It's an `@tmp` path, which means it's the `WORKSPACE_TMP`, which means it should be cleaned up after ever run without use of `cleanWs()`.

Reply:
```
WORKSPACE_TMP
A temporary directory near the workspace that will not be browsable and will not interfere with SCM checkouts. May not initially exist, so be sure to create the directory as needed (e.g., mkdir -p on Linux). Not defined when the regular workspace is a drive root.
```
https://ci.status.im/env-vars.html/

Reply:
The real question is, what is in that folder and what ownership it has. Since ownership are probably the reason why `WORKSPACE_TMP` is not being removed after job finishes like it normally would be.

Reply:
the issue is probably because of `GOPATH`

```
ln: failed to create symbolic link '/home/jenkins/workspace/go_prs_linux_x86_64_main_PR-5907@tmp/go/src/github.com/status-im/status-go': File exists
```
is caused by 
```
 sh "ln -s ${WORKSPACE} ${REPO_SRC}" 
 ```
 
 and `REPO_SRC` is `"${GOPATH}/src/github.com/status-im/status-go"`